sshd/src/main/scala/ammonite/repl/sshd/SshdRepl.scala

```diff
@@ -0,0 +1,78 @@
+package ammonite.repl.sshd
+
+import java.io.{InputStream, OutputStream, PrintStream}
+
+import ammonite.ops.Path
+import ammonite.repl.{Repl, Ref, Storage, Bind}
+import ammonite.repl.sshd.util.Environment
+
+import scala.annotation.tailrec
+import scala.language.postfixOps
+
+/**
+ * An ssh server which serves ammonite repl as it's shell channel. To start listening for incoming connections call
+ * [[start()]] method. You can [[stop()]] the server at any moment. It will also close all running sessions
+ * @param sshConfig configuration of ssh server, such as users credentials or port to be listening to
+ * @param predef predef that will be installed on repl instances served by this server
+ * @param replArgs arguments to pass to ammonite repl on initialization of the session
+ */
+class SshdRepl(sshConfig:SshServerConfig,
+               predef: String = "",
+               replArgs: Seq[Bind[_]] = Nil) {
+  private lazy val sshd = SshServer(sshConfig, terminal = SshdRepl.runRepl(sshConfig.ammoniteHome, predef, replArgs))
+
+  def port = sshd.getPort
+  def start():Unit = sshd.start()
+  def stop():Unit = sshd.stop()
+  def stopImmediately():Unit = sshd.stop(/*immediately =*/ true)
+}
+
+object SshdRepl {
+  def main(args:Array[String]):Unit = {
```

Comment:
This was used for manual standalone testing. I'll move it to the `test` directory.